### PR TITLE
feat: copy handle with domain - WPB-6438

### PR DESCRIPTION
**Pull request by netbe**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-6438" title="WPB-6438" target="_blank">TaskWPB-6438</a>  [iOS] copy handle with domain in account
  </td></table>
  <br />
 

# What's new in this PR?

### Issues

During playtest it is hard to share its handle.
### Causes (Optional)

Can't copy domain in federated backend.

### Solutions

* add long press copy on handle in account
* display full handle with domain in developer tools

### Attachments (Optional)

IMG_6484

### CODE REVIEW

File: wire-ios/Wire-iOS/Sources/UserInterface/Settings/SettingsTableViewController.swift [hunk 1]
```diff
@@ -255,6 +255,37 @@ final class SettingsTableViewController: SettingsBaseTableViewController {
         fatalError("Cannot dequeue cell for index path \(indexPath) and cellDescriptor \(cellDescriptor)")
     }
 
+    func tableView(_ tableView: UITableView, shouldShowMenuForRowAt indexPath: IndexPath) -> Bool {
+        let sectionDescriptor = sections[(indexPath as NSIndexPath).section]
+        let cellDescriptor = sectionDescriptor.visibleCellDescriptors[(indexPath as NSIndexPath).row]
+        return cellDescriptor.canCopy
+    }
+
+    func tableView(_ tableView: UITableView, contextMenuConfigurationForRowAt indexPath: IndexPath, point: CGPoint) -> UIContextMenuConfiguration? {
```
Comment (markusfassbender) [inline wire-ios/Wire-iOS/Sources/UserInterface/Settings/SettingsTableViewController.swift:264]: Nice to use the default `tableView` funcs here 👍

File: wire-ios/Wire-iOS/Sources/UserInterface/Settings/SettingsTableViewController.swift [hunk 2]
```diff
@@ -255,6 +255,37 @@ final class SettingsTableViewController: SettingsBaseTableViewController {
...
+        let sectionDescriptor = sections[(indexPath as NSIndexPath).section]
+        let cellDescriptor = sectionDescriptor.visibleCellDescriptors[(indexPath as NSIndexPath).row]
+
+        guard cellDescriptor.canCopy else {
+            return nil
+        }
+
+        return UIContextMenuConfiguration(identifier: nil, previewProvider: nil) { _ in
+          let copy = UIAction(title: "Copy", image: UIImage(systemName: "doc.on.doc")) { _ in
+            if let cell = tableView.cellForRow(at: indexPath) as? SettingsCellType {
```
Comment (markusfassbender) [inline wire-ios/Wire-iOS/Sources/UserInterface/Settings/SettingsTableViewController.swift]: Do we need to access the cell (View) directly to get the data? Or is there some possibility here in the view controller to access the domain data model?

**Review by markusfassbender — APPROVED** [review]

File: wire-ios/Wire-iOS/Sources/UserInterface/Settings/SettingsTableViewController.swift (hunk 2 above)
Comment (netbe) [reply wire-ios/Wire-iOS/Sources/UserInterface/Settings/SettingsTableViewController.swift]: I can downcast the cellDescriptor to SettingsExternalScreenCellDescriptor and retrieve via the previewGenerator...

or maybe change `canCopy` to `copiableString?` and retrieve from there, I'll try

Comment (netbe) [reply wire-ios/Wire-iOS/Sources/UserInterface/Settings/SettingsTableViewController.swift]: fixed

**Review by johnxnguyen — APPROVED** [review]

**Pull request merged**